snuba_sdk/formula.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, replace
+from enum import Enum
+from typing import Optional, Sequence, Union, Any
+
+from snuba_sdk.aliased_expression import AliasedExpression
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import BooleanCondition, Condition, ConditionGroup
+from snuba_sdk.expressions import Expression, InvalidExpressionError, list_type
+from snuba_sdk.timeseries import Timeseries
+
+
+class InvalidArithmeticError(InvalidExpressionError):
+    pass
+
+
+class InvalidFormulaError(InvalidExpressionError):
+    pass
+
+
+class ArithmeticOperator(Enum):
+    PLUS = "plus"
+    MINUS = "minus"
+    MULTIPLY = "multiply"
+    DIVIDE = "divide"
+
+
+@dataclass(frozen=True)
+class Formula(Expression):
```

Comment:
If this can be modified (with the `set_*`) functions, then it shouldn't extend the Expression class. You'll have to manually call validate on it in the MetricsQuery validator visitor.

Reply:
Removed `Expression` and updated validator visitor to take Formula type.